metaflow/plugins/kubernetes/kubernetes_job.py

```diff
@@ -563,9 +564,29 @@ def has_failed(self):
     @property
     def is_running(self):
         # Returns true if the pod is running.
-        return not self.is_done and (
-            self._pod.get("status", {}).get("phase") == "Running"
-        )
+        if self.is_done:
```

Comment:
the `is_done`, `has_failed` and `has_succeeded` methods need a corresponding update as well, otherwise they will be out of sync and communicate wrong information about the state of the object.